From f6370eec62724dcafedb4181a4fe5960f0bf21b9 Mon Sep 17 00:00:00 2001
Message-Id: <f6370eec62724dcafedb4181a4fe5960f0bf21b9.1427259748.git.chang-joon.lee@intel.com>
In-Reply-To: <1b08866b5b1e0f98fa9e406dc4b29e9df76582fb.1427259748.git.chang-joon.lee@intel.com>
References: <1b08866b5b1e0f98fa9e406dc4b29e9df76582fb.1427259748.git.chang-joon.lee@intel.com>
From: Minu <minu.mathai@intel.com>
Date: Wed, 4 Mar 2015 16:43:33 +0000
Subject: [PATCH 2/2] MUST_REBASE [VPG]: libdrm: adding kernel query param for
 resource streamer

The query param I915_PARAM_HAS_RS is added to i915_get_param. It helps to
detect resource streamer support in kernel at runtime.

MUST_REBASE: This patch can't be upstreamed because there's no open source
use case at present. Mesa are working on adding resource streamer support,
but it's still WIP.

Issue: ACD-134
Change-Id: I679e0f49e47a198a0de18c4b21bf92a58e0f17f5
Signed-off-by: Minu Mathai <minu.mathai@intel.com>
---
 drivers/gpu/drm/i915/i915_dma.c |    3 +++
 drivers/gpu/drm/i915/i915_drv.c |    1 +
 drivers/gpu/drm/i915/i915_drv.h |    2 ++
 include/uapi/drm/i915_drm.h     |    1 +
 4 files changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 72b9dcc..22d917e 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -1041,6 +1041,9 @@ static int i915_getparam(struct drm_device *dev, void *data,
 	case I915_PARAM_HAS_DPST:
 		value = I915_HAS_DPST(dev);
 		break;
+	case I915_PARAM_HAS_RS:
+		value = I915_HAS_RS(dev);
+		break;
 	case I915_PARAM_MMAP_VERSION:
 		value = 1;
 		break;
diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 620f127..7aa476b 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -365,6 +365,7 @@ static const struct intel_device_info intel_cherryview_info = {
 	.ring_mask = RENDER_RING | BSD_RING | BLT_RING | VEBOX_RING,
 	.is_valleyview = 1,
 	.has_dpst = 1,
+	.has_rs = 1,
 	.display_mmio_offset = VLV_DISPLAY_BASE,
 	GEN_CHV_PIPEOFFSETS,
 	CURSOR_OFFSETS,
diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 8534604..3098395 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -579,6 +579,7 @@ struct i915_rpm {
 	func(has_fbc) sep \
 	func(has_pipe_cxsr) sep \
 	func(has_dpst) sep \
+	func(has_rs) sep \
 	func(has_hotplug) sep \
 	func(cursor_needs_physical) sep \
 	func(has_overlay) sep \
@@ -2418,6 +2419,7 @@ enum context_submission_status {
 #define HAS_FBC(dev) (INTEL_INFO(dev)->has_fbc)
 #define I915_HAS_DPST(dev) (INTEL_INFO(dev)->has_dpst && \
 			    !(i915.enable_intel_adf))
+#define I915_HAS_RS(dev) (INTEL_INFO(dev)->has_rs)
 
 #define HAS_IPS(dev)		(IS_ULT(dev) || IS_BROADWELL(dev))
 
diff --git a/include/uapi/drm/i915_drm.h b/include/uapi/drm/i915_drm.h
index 2499dc9..d01095e 100644
--- a/include/uapi/drm/i915_drm.h
+++ b/include/uapi/drm/i915_drm.h
@@ -471,6 +471,7 @@ typedef struct drm_i915_irq_wait {
  */
 #define I915_PARAM_HAS_DPST              0x800
 #define I915_PARAM_EU_COUNT              0x801
+#define I915_PARAM_HAS_RS		 0x802
 
 typedef struct drm_i915_getparam {
 	int param;
-- 
1.7.9.5

